tests/unittests/tests-sixlowpan/tests-sixlowpan.c

```diff
@@ -0,0 +1,265 @@
+{
+    TEST_ASSERT(ng_sixlowpan_nalp(NALP_2));
+}
+
+static void test_sixlowpan_nalp_is_no_6lowpan_frame_3(void)
+{
+    TEST_ASSERT(ng_sixlowpan_nalp(NALP_3));
+}
+
+static void test_sixlowpan_nalp_is_no_6lowpan_frame_4(void)
+{
+    TEST_ASSERT(ng_sixlowpan_nalp(NALP_4));
+}
+
+static void test_sixlowpan_nalp_is_no_6lowpan_frame_5(void)
+{
+    TEST_ASSERT(ng_sixlowpan_nalp(NALP_5));
+}
+
+/* Test with 6LoWPAN dispatch byte indicating some none-NALP value */
+static void test_sixlowpan_nalp_is_6lowpan_frame_0(void)
+{
+    TEST_ASSERT(!ng_sixlowpan_nalp(NO_NALP_0));
+}
+
+static void test_sixlowpan_nalp_is_6lowpan_frame_1(void)
+{
+    TEST_ASSERT(!ng_sixlowpan_nalp(NO_NALP_1));
+}
+
+static void test_sixlowpan_nalp_is_6lowpan_frame_2(void)
+{
+    TEST_ASSERT(!ng_sixlowpan_nalp(NO_NALP_2));
+}
+
+static void test_sixlowpan_nalp_is_6lowpan_frame_3(void)
+{
+    TEST_ASSERT(!ng_sixlowpan_nalp(NO_NALP_3));
+}
+
+static void test_sixlowpan_nalp_is_6lowpan_frame_4(void)
+{
+    TEST_ASSERT(!ng_sixlowpan_nalp(NO_NALP_4));
+}
+
+static void test_sixlowpan_nalp_is_6lowpan_frame_5(void)
+{
+    TEST_ASSERT(!ng_sixlowpan_nalp(NO_NALP_5));
+}
+
+/* Test with 6LoWPAN dispatch byte indicating a LoWPAN frame */
+static void test_sixlowpan_nalp_is_6lowpan_frame_6(void)
+{
+    TEST_ASSERT(!ng_sixlowpan_nalp(IPv6_DISP));
+}
+
+static void test_sixlowpan_nalp_is_6lowpan_frame_7(void)
+{
+    TEST_ASSERT(!ng_sixlowpan_nalp(LOWPAN_HC1_DISP));
+}
+
+static void test_sixlowpan_nalp_is_6lowpan_frame_8(void)
+{
+    TEST_ASSERT(!ng_sixlowpan_nalp(LOWPAN_BC0_DISP));
+}
+
+static void test_sixlowpan_nalp_is_6lowpan_frame_9(void)
+{
+    TEST_ASSERT(!ng_sixlowpan_nalp(ESC_DISP));
+}
+
+static void test_sixlowpan_nalp_is_6lowpan_frame_10(void)
+{
+    TEST_ASSERT(!ng_sixlowpan_nalp(MESH_DISP));
+}
+
+static void test_sixlowpan_nalp_is_6lowpan_frame_11(void)
+{
+    TEST_ASSERT(!ng_sixlowpan_nalp(FRAG1_DISP));
+}
+
+static void test_sixlowpan_nalp_is_6lowpan_frame_12(void)
+{
+    TEST_ASSERT(!ng_sixlowpan_nalp(FRAGN_DISP));
+}
+
+/* Test initialization of 6LoWPAN */
+static void test_sixlowpan_init_success(void)
+{
+    kernel_pid_t sl_pid = KERNEL_PID_UNDEF;
+    sl_pid = ng_sixlowpan_init();
+    TEST_ASSERT(sl_pid != -EOVERFLOW);
+    TEST_ASSERT(sl_pid != -EINVAL);
+    TEST_ASSERT(sl_pid != KERNEL_PID_UNDEF);
+}
+
+static void test_sixlowpan_init_twice(void)
+{
+    kernel_pid_t sl_pid1 = KERNEL_PID_UNDEF;
+    kernel_pid_t sl_pid2 = KERNEL_PID_UNDEF;
+    sl_pid1 = ng_sixlowpan_init();
+    TEST_ASSERT(sl_pid1 != -EOVERFLOW);
+    TEST_ASSERT(sl_pid1 != -EINVAL);
+    TEST_ASSERT(sl_pid1 != KERNEL_PID_UNDEF);
+    sl_pid2 = ng_sixlowpan_init();
+    TEST_ASSERT_EQUAL_INT(sl_pid1, sl_pid2);
+}
+
+static void test_sixlowpan_init_invalid_prio(void)
+{
+#undef NG_SIXLOWPAN_PRIO
+#define NG_SIXLOWPAN_PRIO   (SCHED_PRIO_LEVELS + 1)
+    kernel_pid_t sl_pid = KERNEL_PID_UNDEF;
+    sl_pid = ng_sixlowpan_init();
+    TEST_ASSERT_EQUAL_INT(-EINVAL, sl_pid);
```

Comment:
Will never actually be tested (see https://github.com/RIOT-OS/RIOT/pull/3307/files#r33860255)